### PR TITLE
vsphere: vmdk-to-ovf.sh: add an option to us legacy bios for podvm te…

### PR DESCRIPTION
**Pull request by whitebrandy**
…mplate

CentOS cloud images do not support EFI. Templates converted from CentOS podvm qcow2s are unable to boot on esx. Add a switch "-disableefi" that can be used as such:

./vmdk-to-ovf.sh <vCenter Server> <vCenter Username> <vCenter Password>...--disableefi

Fixes:#527
Signed-off-by: Bandan Das <bsd@redhat.com>

### CODE REVIEW

**Comment by whitebrandy** [comment]
Note that this depends on #511

**Review by bpradipt — APPROVED** [review]
/lgtm

File: vsphere/vmdk-to-ovf.sh [hunk 1]
```diff
@@ -31,6 +31,12 @@ TEMPLATE_NAME=$6
 # Path to the vmdk to convert to a template
 VMDK_PATH=$7
 
+# Legacy bios or EFI
+LEGACY_BIOS=${8:--notset}
+LEGACYBIOS=0
+[ "$LEGACY_BIOS" = "-disableefi" ] && LEGACYBIOS=1
```
Comment (littlejawa) [inline vsphere/vmdk-to-ovf.sh:37]: The "usage" asks for "--disableefi", but we test for a single "-"

File: vsphere/vmdk-to-ovf.sh [hunk 2]
```diff
@@ -7,8 +7,8 @@ NOCOLOR='\033[0m'
 [ "$DEBUG" == 'true' ] && set -x
 
 ARGC=$#
-if [ $ARGC -ne 7 ]; then
-    echo "USAGE: $(basename $0) <vCenter Server> <vCenter Username> <vCenter Password> <vCenter Cluster> <vCenter Datastore> <Template Name> <Path to VMDK>"
+if [ $ARGC -lt 7 ]; then
```
Comment (littlejawa) [inline vsphere/vmdk-to-ovf.sh:10]: If we miss a parameter but provide the optional parameter, we end up with VMDK_PATH set to "--disableefi"
Obviously, this should cause a visible error pretty quick, but maybe we can check for it right away and prevent issues?

```suggestion
if [ $ARGC -lt 7 || "$7" = "--disableefi"]; then
```

File: vsphere/vmdk-to-ovf.sh (hunk 1 above)
Comment (whitebrandy) [reply vsphere/vmdk-to-ovf.sh:37]: Fixed.

File: vsphere/vmdk-to-ovf.sh [hunk 3]
```diff
@@ -7,8 +7,8 @@ NOCOLOR='\033[0m'
 [ "$DEBUG" == 'true' ] && set -x
 
 ARGC=$#
-if [ $ARGC -ne 7 ]; then
-    echo "USAGE: $(basename $0) <vCenter Server> <vCenter Username> <vCenter Password> <vCenter Cluster> <vCenter Datastore> <Template Name> <Path to VMDK>"
+if [ $ARGC -lt 7 ] || [ "$7" = "-disableefi" ]; then
+    echo "USAGE: $(basename $0) <vCenter Server> <vCenter Username> <vCenter Password> <vCenter Cluster> <vCenter Datastore> <Template Name> <Path to VMDK> <-disableefi>(Optional)"
```
[the remark this answers is not in the document]
Comment (littlejawa) [reply vsphere/vmdk-to-ovf.sh:11]: More checks are always good, but I think this is good enough for now :-)

**Review by littlejawa — APPROVED** [review]
lgtm
Thanks @whitebrandy

**Pull request merged**